packages/@webex/webex-core/src/lib/credentials/credentials.js

```diff
@@ -258,8 +258,28 @@ const Credentials = WebexPlugin.extend({
    * @returns {Promise<Token>}
    */
   downscope(scope) {
+    const superTokenScopeArr = this.supertoken.scope.split(' ');
+    const scopeArr = scope.split(' ');
+
+    const scopeDiff = difference(scopeArr, superTokenScopeArr);
+    const newScope = without(scopeArr, ...scopeDiff).join(' ');
+
```

Comment:
I think it would be better to do this earlier, so that the scopes are stored correctly, and error cases don't keep happening. Suggest:
- in credentials refresh: check each token if it is a subset of the supertoken. If not, just revoke and remove.
- in credentials getUserToken, change the default from the config scope minus spark:kms to the supertoken scope minus spark:kms
- keep this function as it was, but with the improved log message if defaulting to supertoken (which we would no longer expect to see, so maybe make it a warning?) (we could add a metric here maybe?)
- in credentials refresh: just after refreshing the supertoken, compare its scope to the config scope, and emit a warning (and ideally a metric) if the scope is different (this could be a different PR) (this is where we would intend to force logout)

Reply:
@lismith2-cisco 
Is it necessary? In the `refresh` method after the super token refresh, we revoke all user tokens and fetch a new one. 

I could filter out invalid scopes here, but the downscope method will do it anyway.

Reply:
as far as I can see, if we have an invalid scope, we will fall back to supertoken, with the stored scope of the token object being the invalid scope still

Reply:
the only case when it is matter when the user update the app to the version which contains these fixes.

Reply:
Yes, but that is all existing users, and it will happen every time they refresh from now on

Reply:
sorry, I thought you will see the fix.

I agree and it was fixed in the last commit.